### PR TITLE
[GPU] Support feature broadcasting in eltwise vload8

### PR DESCRIPTION
**Pull request by dhandhalyabhavik**
## FPS Gain:
**2.48%**

### Details:

- Extend `eltwise_simple_vload8` to support a conservative feature-only broadcast from `[B, 1, Y, X]` to `[B, F, Y, X]`.
- Compute the broadcast input vector index from the output batch and spatial-vector coordinates while reusing the single feature plane.
- Size dispatch from the output tensor, which also fixes the existing scalar-first input ordering case.
- Keep the optimized path restricted to static, plain `bfyx` tensors with zero offsets, no padding/pitch differences, matching batch/Y/X dimensions, and a spatial plane divisible by 8.
- Preserve existing kernel selection for same-shape and scalar inputs. Unsupported layouts, alignment, and other broadcast dimensions continue through the existing fallback path.
- Add focused selector and byte-exact differential tests for FP16, FP32, BF16, both operand orders, batch > 1, same-shape/scalar regressions, and six fallback boundaries.

This PR intentionally contains only the vload8 selector/JIT change and its tests.

### Validation:

- Release Intel GPU plugin and `ov_gpu_unit_tests` build: passed.
- Focused vload8 tests: **14/14 passed**.
- Eltwise regression suite: **164/164 passed**.
- Test-only stock baseline: **7 expected positive/dispatch selection failures and 7 existing/fallback passes**. Stock selected `generic_eltwise_ref` for feature-broadcast cases.
- GroundingDINO dynamic model checks at B1/L6, B1/L8, and B2/L6: logits and boxes are byte-identical to stock.
- Kernel profiling: all six target encoder multiplies select `eltwise_simple_vload8`; no weighted-reduce change is included.
- B60-dGPU and PTL iGPU were used for testing, batch 1/text length 6, 3 × 30-second runs, 4 infer requests:
  - gain: **+2.48%**
- `git diff --check`, copyright, clang-format-18, and changed-line clang-tidy-18: passed.

### Tickets:

CVS-194431

### AI Assistance:

- **AI assistance used: yes.**
- AI was used to help analyze profiling data, suggest implementation and edge-case tests, review code paths, diagnose test failures, and draft the PR description.
- Human validation performed: contributors defined and reviewed the intended selector/fallback behavior; the change was rebuilt on current upstream `master`; focused and regression tests, stock-versus-patched selector checks, byte-exact model comparisons, kernel profiling, formatting/static-analysis checks, and repeated performance measurements were run locally. The contributors take responsibility for the submitted change.

### CODE REVIEW

**Comment by deepaks2** [comment]
build_jenkins

File: src/plugins/intel_gpu/src/kernel_selector/kernels/eltwise/eltwise_kernel_vload8.cpp [hunk 1]
```diff
@@ -21,11 +39,44 @@ ParamsKey EltwiseKernel_vload8::GetSupportedKey() const {
...
-    return GetJitConstantsCommon(params, true);
+    auto jit = GetJitConstantsCommon(params, true);
+
+    const auto& output = params.outputs[0];
+    const bool has_feature_broadcast = std::any_of(params.inputs.begin(), params.inputs.end(), [&](const DataTensor& input) {
+        return IsFeatureBroadcast(input, output);
+    });
+    if (!has_feature_broadcast)
+        return jit;
+
```
Comment (deepaks2) [inline src/plugins/intel_gpu/src/kernel_selector/kernels/eltwise/eltwise_kernel_vload8.cpp:55]: Duplicated VLOAD_DECLS construction. Iooks like the scalar/buffer loop copied from EltwiseKernelBase::MakeLoadJitConstants, then removed and re-added. Consider extending the base helper (e.g., a protected overload or an index-resolver callback that knows the broadcast case) so all three load modes (scalar, plain, feature-broadcast) are generated in one place.

File: src/plugins/intel_gpu/src/kernel_selector/kernels/eltwise/eltwise_kernel_vload8.cpp [hunk 2]
```diff
@@ -72,12 +123,15 @@ bool EltwiseKernel_vload8::Validate(const Params& params) const {
...
-        if (ewParams.inputs[i].PitchesDifferFromLogicalDims() ||
-            ((ewParams.inputs[0] != ewParams.inputs[i] || ewParams.inputs[i] != ewParams.outputs[0]) &&
-             ewParams.inputs[i].PhysicalSize() != 1)) {
+        // Allow equal-sized inputs, scalars, or a plain bfyx input which differs
+        // only by broadcasting feature=1 to the output feature count.
+        const bool same_as_output = ewParams.inputs[i] == output;
+        const bool scalar = ewParams.inputs[i].PhysicalSize() == 1;
+        const bool feature_broadcast = !ewParams.is_shape_agnostic && IsFeatureBroadcast(ewParams.inputs[i], output);
+        if ((!same_as_output && !scalar && !feature_broadcast) || ewParams.inputs[i].PitchesDifferFromLogicalDims() ||
+            ewParams.inputs[i].GetFirstElementOffset() != 0) {
```
Comment (deepaks2) [inline src/plugins/intel_gpu/src/kernel_selector/kernels/eltwise/eltwise_kernel_vload8.cpp:134]: Padded-output rejection is a behavior change was the old path reachable? —  Validate ,  bCheckSizes?? 
I think the new bCheckSizes initializer rejecting output pitch/first-element-offset. Can you check?

File: src/plugins/intel_gpu/src/kernel_selector/kernels/eltwise/eltwise_kernel_vload8.cpp [hunk 3]
```diff
@@ -72,12 +123,15 @@ bool EltwiseKernel_vload8::Validate(const Params& params) const {
...
     for (size_t i = 0; i < ewParams.inputs.size(); i++) {
-        // allow only the same input sizes or scalars, without pitches
-        if (ewParams.inputs[i].PitchesDifferFromLogicalDims() ||
-            ((ewParams.inputs[0] != ewParams.inputs[i] || ewParams.inputs[i] != ewParams.outputs[0]) &&
-             ewParams.inputs[i].PhysicalSize() != 1)) {
+        // Allow equal-sized inputs, scalars, or a plain bfyx input which differs
+        // only by broadcasting feature=1 to the output feature count.
+        const bool same_as_output = ewParams.inputs[i] == output;
+        const bool scalar = ewParams.inputs[i].PhysicalSize() == 1;
+        const bool feature_broadcast = !ewParams.is_shape_agnostic && IsFeatureBroadcast(ewParams.inputs[i], output);
```
Comment (deepaks2) [inline src/plugins/intel_gpu/src/kernel_selector/kernels/eltwise/eltwise_kernel_vload8.cpp:132]: Do we gave tests for is_shape_agnostic guard??

**Review by alvoron — CHANGES_REQUESTED** [review]
Agree with all 3 comments of @deepaks2 
@dhandhalyabhavik  could you please address them?

File: src/plugins/intel_gpu/src/kernel_selector/kernels/eltwise/eltwise_kernel_vload8.cpp (hunk 1 above)
Comment (dhandhalyabhavik) [reply src/plugins/intel_gpu/src/kernel_selector/kernels/eltwise/eltwise_kernel_vload8.cpp:56]: Thanks, addressed in `a66cd4e`. `VLOAD_DECLS` is now generated only in `EltwiseKernelBase::MakeLoadJitConstants()`. I added a protected `GetVload8InputIndex()` hook whose default returns `global_id`; `EltwiseKernel_vload8` overrides only the feature-broadcast index. Scalar, ordinary-buffer, and feature-broadcast loads now share one construction path.

File: src/plugins/intel_gpu/src/kernel_selector/kernels/eltwise/eltwise_kernel_vload8.cpp (hunk 2 above)
Comment (dhandhalyabhavik) [reply src/plugins/intel_gpu/src/kernel_selector/kernels/eltwise/eltwise_kernel_vload8.cpp]: Thanks for pointing this out. I restored the legacy `bCheckSizes = true` initialization. Layouts requiring tensor-pitch or tensor-offset support are already rejected through the vload8 `ParamsKey`, while `IsFeatureBroadcast()` remains strict for the new broadcast path. I also aligned validation with dispatch by checking `output.LogicalSize() % 8`, because work-group dispatch uses `LogicalSize() / 8`. The tests now cover both boundaries: logical size 10 with physical size 40 correctly falls back, while logical size 8 with physical size 9 selects vload8 and remains byte-exact.

File: src/plugins/intel_gpu/src/kernel_selector/kernels/eltwise/eltwise_kernel_vload8.cpp (hunk 3 above)
Comment (dhandhalyabhavik) [reply src/plugins/intel_gpu/src/kernel_selector/kernels/eltwise/eltwise_kernel_vload8.cpp:111]: Added the direct selector-level test `vload8_rejects_shape_agnostic_feature_broadcast`. It constructs an otherwise eligible feature-broadcast candidate and changes only `is_shape_agnostic`: validation rejects it when `true` and accepts the same candidate when `false`. This makes the test specifically sensitive to the guard. Simply if shape known at compile then vload8 kernel gets selected otherwise generic.

**Comment by dhandhalyabhavik** [comment]
> Agree with all 3 comments of @deepaks2 @dhandhalyabhavik could you please address them?

Done, can you please re-review?

File: src/plugins/intel_gpu/src/kernel_selector/kernels/eltwise/eltwise_kernel_vload8.cpp [hunk 4]
```diff
@@ -2,14 +2,32 @@
...
+bool IsFeatureBroadcast(const DataTensor& input, const DataTensor& output) {
+    if (input.GetLayout() != DataLayout::bfyx || output.GetLayout() != DataLayout::bfyx)
+        return false;
+
+    if (input.PitchesDifferFromLogicalDims() || output.PitchesDifferFromLogicalDims() || input.GetFirstElementOffset() != 0 ||
+        output.GetFirstElementOffset() != 0)
+        return false;
+
+    return input.Batch().v == output.Batch().v && input.Feature().v == 1 && output.Feature().v > 1 && input.Y().v == output.Y().v &&
+           input.X().v == output.X().v && (input.Y().v * input.X().v) % 8 == 0;
```
[the remark this answers is not in the document]
Comment (alvoron) [reply src/plugins/intel_gpu/src/kernel_selector/kernels/eltwise/eltwise_kernel_vload8.cpp]: @dhandhalyabhavik 
I agree with the comment and with the fix. Could you please review it and apply it if it looks good to you?

**Review by alvoron — APPROVED** [review]

File: src/plugins/intel_gpu/src/kernel_selector/kernels/eltwise/eltwise_kernel_vload8.cpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (dhandhalyabhavik) [reply src/plugins/intel_gpu/src/kernel_selector/kernels/eltwise/eltwise_kernel_vload8.cpp]: Yes, fix added.

**Review by deepaks2 — APPROVED** [review]

**Comment by deepaks2** [comment]
build_jenkins

**Comment by p-durandin** [comment]
build_jenkins
